app/frontend/src/components/autocomplete/autocomplete.js

```diff
@@ -6,40 +7,47 @@ import api from '../../lib/api';
 
 const SHOW_SUGGESTIONS_THRESHOLD = 3;
 
-const highlightRefinement = (text, refinement) => {
-  const index = text.toLowerCase().indexOf(refinement.toLowerCase());
+const suggestionHTML = (text, inputValue) => {
+  const index = text.toLowerCase().indexOf(inputValue.toLowerCase());
 
   /* eslint-disable max-len */
-  return `${text.substring(0, index)}<span class='accessible-autocomplete__suggestion-highlight'>${text.substring(index, index + refinement.length)}</span>${text.substring(index + refinement.length, text.length)}`;
+  return `${text.substring(0, index)}<span class='accessible-autocomplete__suggestion-highlight'>${text.substring(index, index + inputValue.length)}</span>${text.substring(index + inputValue.length, text.length)}`;
   /* eslint-enable */
 };
 
-const autocomplete = () => {
-  Array.from(document.querySelectorAll('.autocomplete')).forEach((el) => {
-    const formInput = el.querySelector('input');
-    const dataSource = api[el.dataset.source];
+export default class extends Controller {
+  connect() {
+    const formInput = this.element.querySelector('input');
+    const dataSource = api[this.element.dataset.source];
+    const position = this.element.dataset.autocompletePosition;
+
+    this.addContainers();
 
     if (formInput && dataSource) {
       let currentInputValue = formInput.value;
       formInput.parentNode.removeChild(formInput);
 
       accessibleAutocomplete({
-        element: el.querySelector('.accessible-autocomplete'),
+        element: this.element.querySelector('.accessible-autocomplete'),
```

Comment:
and for this?

Reply:
similar, had a quandry, but i think it should be internal to the autocomplete js. its an odd one this as stimulus doesnt bring much to the party in this case. all we are doing is encapsulating govuk-accessible-autocomplete so i dont thing need to over engineer it. good spot pointing these out, i did give it a bit of thought but decided against. having said that i did think some more with your comments and have made things a little tighter by removing some uneccessary mark up and styles

Reply:
having said that i did think some more because of your comment and have made things a little tighter by removing some uneccessary mark up and styles which i have updated this PR with